src/python/aqueduct_executor/operators/connectors/data/extract.py

```diff
@@ -16,15 +18,66 @@ class RelationalParams(models.BaseParams):
     # any user-defined tags like `{{today}}`.
     query_is_usable: Optional[bool] = False
 
-    query: str
+    query: Optional[str] = None
+    queries: Optional[List[str]] = None
 
     # TODO: Consider not including github as part of relational params when it is JSON marshalled
     github_metadata: Optional[Any]
 
-    def expand_placeholders(
-        self,
-        parameters: Dict[str, str],
-    ) -> None:
+    def _compile_chain(self, queries: List[str]) -> str:
+        """
+        `_compile_chain` compiles a chain query to a single query using `WITH` clause.
+        We generate temp_table_name and replace PREV_TABLE_TAG accordingly.
+
+        Example:
+        queries: [
+            "SELECT * FROM my_table",
+            "SELECT field_a, field_b FROM $",
+            "SELECT * FROM $",
+        ]
+
+        returns: `
+            WITH
+                generated_tmp_a AS (SELECT * FROM my_table),
+                generated_tmp_b AS (SELECT field_a, field_b FROM generated_tmp_a)
+            SELECT * FROM generated_tmp_b
+        `
+        """
+        if not queries:
+            return ""
+
+        if len(queries) > 1:
+            with_clause = "WITH\n"
+            prev_table_name = ""
+            for (idx, query) in enumerate(queries):
+                # remove spaces and trailing semicolumns if any.
+                normalized_query = query.strip().rstrip(";")
+
+                # replace tag except for the first query
+                if idx == 0:
+                    if PREV_TABLE_TAG in normalized_query:
+                        raise Exception(
+                            f"Cannot compile chain. {PREV_TABLE_TAG} appears in the first query: {query}"
+                        )
+                else:
+                    normalized_query = normalized_query.replace(PREV_TABLE_TAG, prev_table_name)
+
+                # subquery goes to the 'WITH' clause except for the last one.
+                if idx < len(queries) - 1:
+                    cur_table_name = f"aqueduct_{uuid.uuid4().hex}"
+                    with_clause += f"{cur_table_name} AS (\n{normalized_query}\n)"
+
+                    # there are more subqueries to append in this 'WITH' clause
+                    if idx < len(queries) - 2:
+                        with_clause += ",\n"
+                    prev_table_name = cur_table_name
+                # otherwise, append the last query to 'WITH' clause
+                else:
+                    return f"{with_clause}\n{normalized_query}"
+
+        return queries[0]
```

Comment:
Can we put this in an else statement? This prevents the if statement above from forgetting to return something.

Reply:
I decided to swap so that we have `if len(queries)==1: return queries[0]`